python/ufoff.py

```diff
@@ -0,0 +1,114 @@
+
+from ufoLib2 import Font
+import flatbuffers
+from FlatFont.Ufo import FontInfo
+from enum import IntEnum
+
+SCALARS = [int, float, bool] # Other scalars needed # Enum needed
+NON_SCALARS = [str, list] # Other none-scalars needed
+
+def Capital(string: str): # seek equivalent buit-in function
+    '''Capitalizes the string without affecting other characters'''
+    return string[0].upper() + string[1:]
+
+def StartAttrVector(attr: str):
+    '''Simplifies calling buffer functions by returning a string representing the buffer function suitable for the "Getattr" function input'''
+    return "Start" + Capital(attr) + "Vector"
+
+def AddAttr(attr: str): # how about make it a subclass to str, add a method with or without @property: attr.AddAttr or attr.StartAttrVector
+    '''Simplifies calling buffer functions by returning a string representing the buffer function suitable for the "Getattr" function input'''
+    return "Add" + Capital(attr)
+
+def BuildAttrString(builder: flatbuffers.Builder, string: str):
+    '''Creates a buffer for strings'''
+    return builder.CreateString(string)
+
+def BuildAttrVector(builder: flatbuffers.Builder, attr: str ,vector: list, Value_Type: type = float):   
+    '''Creates a buffer for Vectors by passing the name of the attribute as a string'''
+    getattr(FontInfo, StartAttrVector(attr))(builder, len(vector))
+    for index in reversed(range(0, len(vector))):
+        if Value_Type == float:
+            builder.PrependFloat32(vector[index])
+        if Value_Type == int:
+            builder.PrependInt32(vector[index])
+        if Value_Type == bool:
+            builder.PrependBool(vector[index])
+    return builder.EndVector()
+
+
+
+
+def GetAttrReference(builder: flatbuffers.Builder, ufo: Font) -> dict: 
+    '''
+    Creates a dictionary  that contains references for each data type that needs the buffer
+    Includes a for-loop to generate a reference for the buffer-needed fields
+    To avoid NestedError
+    '''
+    attr_dict = dict()
+    # A for-loop to create a reference to all fields which needs the buffer to be created for avoiding NestedError
+    for attr in dir(ufo.info):
```

Comment:
don't do `dir(ufo.info)`, you'll get a bunch of private attributes prefixed with "_" that are actually implemented as property getters/setters in ufoLib2.
The correct way to convert a ufoLib2 object (any object including a Font) to a json-like dictionary is to use the ufoLib2.converters.unstructure method. For that you need to `pip install ufoLib2[converters]` because it's an extra (optional) feature that depends on third-party `cattrs` library.
then you can do `from ufoLib2.converters import unstructure; info_dict = unstructure(font.info)`

Reply:
BTW, there is also a `structure` method that does the reverse of that, i.e. takes in a json-like dict and a ufoLib2 class and creates an object with that data (recursively).